src/main/java/technology/dice/dicefairlink/discovery/members/FairlinkMemberFinder.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (C) 2018 - present by Dice Technology Ltd.
+ *
+ * Please see distribution for license.
+ */
+package technology.dice.dicefairlink.discovery.members;
+
+import technology.dice.dicefairlink.config.FairlinkConfiguration;
+import technology.dice.dicefairlink.discovery.tags.ExclusionTag;
+import technology.dice.dicefairlink.discovery.tags.TagFilter;
+import technology.dice.dicefairlink.driver.FairlinkConnectionString;
+import technology.dice.dicefairlink.iterators.SizedIterator;
+
+import java.net.URISyntaxException;
+import java.time.Duration;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+public class FairlinkMemberFinder {
+  private static final Logger LOGGER = Logger.getLogger(FairlinkMemberFinder.class.getName());
+  private static final ExclusionTag EXCLUSION_TAG = new ExclusionTag("Fairlink-Exclude", "true");
+  private static final Set<String> EMPTY_SET = new HashSet<>(0);
```

Comment:
`ImmutableSet.of()`

Reply:
This project doesn't have a dependency on Guava (only on test scope)

Reply:
probably worth wrapping in the `Collections.unmodifiable...`